src/Backups/BackupsWorker.cpp

```diff
@@ -659,11 +662,16 @@ void BackupsWorker::doBackup(
             setStatusSafe(backup_id, getBackupStatusFromCurrentException());
             sendCurrentExceptionToCoordination(backup_coordination);
         }
-        else
+
+        if (backup)
```

Comment:
Maybe its better to have this `if (backup)` block before checking ` if (called_async)`, and keep the `called_async` if else as it was before. Currently if its async it will throw exception and no remove files, correct me if I am wrong here.

Reply:
It's better to show a log message and update the status in `system.backups` before removing files. Because sometimes that removing can be slow or even unsuccessful.

I simplified the code around here a bit - checking of `called_async` wasn't really necessary here and `scheduleFromThreadPool()` can do the stuff related to thread groups.